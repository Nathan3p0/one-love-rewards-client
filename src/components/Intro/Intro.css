.main__content-intro {
    background: #CF5C36;
    display: flex;
    justify-content: center;
    align-items: center;
    text-align: center;
}

.main__content-intro--logo {
    max-width: 30rem;
}

.main__content-intro--container h2{
    font-size: 2rem;
    color: #EEE5E9;
    font-weight: 400;
    font-family: 'Playfair Display', serif;
}

button {
    position: relative;
    display: inline-block;
    cursor: pointer;
    outline: none;
    border: 0;
    vertical-align: middle;
    text-decoration: none;
    font-size: inherit;
    font-family: inherit;
  }
  button.btn {
    transition: all 0.5s cubic-bezier(0.65, -0.25, 0.25, 1.95);
    font-weight: 900;
    color: #202121;
    padding: 1.25rem 2rem;
    background: #fff;
    text-transform: uppercase;
  }
  button.btn:hover, button.btn:focus, button.btn:active {
    letter-spacing: 0.125rem;
  }

/*
.customer__btn {
    padding: 1rem;
    margin-right: .5rem;
    background-color: #000000;
    color: #fff;
} */

.business__btn {
    padding: 1rem;
    margin-left: .5rem;
    background-color: #000000;
    color: #fff;
}


@media screen and (max-width: 800px) {
    .main__content-intro {

        align-items: flex-start;
        height: 90vh;
    }

    .main__content-intro--logo {
        max-width: 25rem;
    }
}