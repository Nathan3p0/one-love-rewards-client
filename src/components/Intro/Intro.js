import React from 'react'
import largeLogo from './images/large-logo.png'
import './Intro.css'

const Intro = (props) => {
    return (
        <section className="main__content-intro">
            <div className="main__content-intro--container">
                <img src={largeLogo} alt="" className="main__content-intro--logo" />
                <h2>REWARDS MADE EASY</h2>
                <button onClick={props.handleCustomerLoginToggle} className="btn customer__btn">Customer Login</button>
                <button onClick={props.handleBusinessLoginToggle} className="btn business__btn">Business Login</button>
            </div>
        </section>
    )
}

export default Intro