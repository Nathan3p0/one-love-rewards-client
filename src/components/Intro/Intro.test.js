import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Intro from './Intro'

describe('Intro Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<Intro />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<Intro />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<Intro
            handleBusinessLoginToggle={() => { }}
            handleCustomerLoginToggle={() => { }}
        />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})