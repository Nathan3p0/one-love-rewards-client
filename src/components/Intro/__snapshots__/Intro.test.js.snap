// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Component Rendering Tests renders to UI as expected 1`] = `
<section
  className="main__content-intro"
>
  <div
    className="main__content-intro--container"
  >
    <img
      alt=""
      className="main__content-intro--logo"
      src="large-logo.png"
    />
    <h2>
      Dining Rewards Made Easy...
    </h2>
    <button
      className="btn customer__btn"
      onClick={[Function]}
    >
      Customer Login
    </button>
    <button
      className="btn business__btn"
      onClick={[Function]}
    >
      Business Login
    </button>
  </div>
</section>
`;

exports[`Intro Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<section
  className="main__content-intro"
>
  <div
    className="main__content-intro--container"
  >
    <img
      alt=""
      className="main__content-intro--logo"
      src="large-logo.png"
    />
    <h2>
      Dining Rewards Made Easy...
    </h2>
    <button
      className="btn customer__btn"
      onClick={[Function]}
    >
      Customer Login
    </button>
    <button
      className="btn business__btn"
      onClick={[Function]}
    >
      Business Login
    </button>
  </div>
</section>
`;
